CHANGELOG.md

```diff
@@ -1,3 +1,274 @@
+- Pass client information on LSP initialization (#4904)
+- Allow specifying environment variables for language servers in language config (#4004)
+- Allow detached git worktrees to be recognized as root paths (#5097)
+- Improve error message handling for theme loading failures (#5073)
+- Print the names of binaries required for LSP/DAP in health-check (#5195)
+- Improve sorting in the picker in cases of ties (#5169)
+- Add theming for prompt suggestions (#5104)
+- Open a file picker when using `:open` on directories (#2707, #5278)
+- Reload language config with `:config-reload` (#5239, #5381, #5431)
+- Improve indent queries for python when the tree is errored (#5332)
+- Picker: Open files without closing the picker with `A-ret` (#4435)
+- Allow theming cursors by primary/secondary and by mode (#5130)
+- Allow configuration of the minimum width for the line-numbers gutter (#4724, #5696)
+- Use filename completer for `:run-shell-command` command (#5729)
+- Surround with line-endings with `ms<ret>` (#4571)
+- Hide duplicate symlinks in file pickers (#5658)
+- Tabulate buffer picker contents (#5777)
+- Add an option to disable LSP (#4425)
+- Short-circuit tree-sitter and word object motions (#5851)
+- Add exit code to failed command message (#5898)
+- Make `m` textobject look for pairs enclosing selections (#3344)
+- Negotiate LSP position encoding (#5894)
+- Display deprecated LSP completions with strikethrough (#5932)
+- Add JSONRPC request ID to failed LSP/DAP request log messages (#6010, #6018)
+- Ignore case when filtering LSP completions (#6008)
+- Show current language when no arguments are passed to `:set-language` (#5895)
+- Refactor and rewrite all book documentation (#5534)
+- Check LSP renaming support before prompting for rename text (#6257)
+- Separate diagnostic picker message and code (#6095)
+- Add a config option to bypass undercurl detection (#6253)
+- Only complete appropriate arguments for typed commands (#5966)
+- Discard outdated LSP diagnostics (3c9d5d0)
+- Discard outdated LSP workspace edits (b6a4927)
+- Run shell commands asynchronously (#6373)
+- Show diagnostic codes in LSP diagnostic messages (#6378)
+
+Fixes:
+
+- Fix behavior of `auto-completion` flag for completion-on-trigger (#5042)
+- Reset editor mode when changing buffers (#5072)
+- Respect scrolloff settings in mouse movements (#5255)
+- Avoid trailing `s` when only one file is opened (#5189)
+- Fix erroneous indent between closers of auto-pairs (#5330)
+- Expand `~` when parsing file paths in `:open` (#5329)
+- Fix theme inheritance for default themes (#5218)
+- Fix `extend_line` with a count when the current line(s) are selected (#5288)
+- Prompt: Fix autocompletion for paths containing periods (#5175)
+- Skip serializing JSONRPC params if params is null (#5471)
+- Fix interaction with the `xclip` clipboard provider (#5426)
+- Fix undo/redo execution from the command palette (#5294)
+- Fix highlighting of non-bar cursors (#5575)
```

Comment:
This is actually a typo in the PR title it should be non-bkock cursors not non-bar cursors